File: benchmarks/linux/process_bench.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#include <unistd.h>
#include <sys/wait.h>

#define ITERATIONS 20

int main() {
    struct timespec start, end;
    double total = 0.0;

    for (int i = 0; i < ITERATIONS; i++) {
        clock_gettime(CLOCK_MONOTONIC, &start);

        pid_t pid = fork();
        if (pid == 0) {
            execl("/bin/true", "true", NULL);
            _exit(1);
        } else if (pid < 0) {
            perror("fork");
            return 1;
        }

        int status;
        waitpid(pid, &status, 0);

        clock_gettime(CLOCK_MONOTONIC, &end);
        double elapsed = (end.tv_sec - start.tv_sec) * 1000.0
                       + (end.tv_nsec - start.tv_nsec) / 1e6;
        total += elapsed;
    }

    printf("RESULT:%.4f\n", total / ITERATIONS);
    return 0;
}

File: benchmarks/linux/process_bench.py
"""Linux process-creation benchmark: fork() + execl() + waitpid()."""
import os
import sys
import time

ITERATIONS = 20
total      = 0.0

for _ in range(ITERATIONS):
    t0  = time.perf_counter()
    pid = os.fork()
    if pid == 0:
        os.execl('/bin/true', 'true')
        os._exit(1)
    if pid < 0:
        print('fork failed', file=sys.stderr)
        sys.exit(1)
    os.waitpid(pid, 0)
    t1 = time.perf_counter()
    total += t1 - t0

print(f'RESULT:{(total / ITERATIONS) * 1000:.4f}')
